### PR TITLE
Replace `MemberAdaptor::fetch_by_stable_id` in examples

### PR DESCRIPTION
**Pull request by CristiGuijarro**
## Description

There are a few scripts that require an update upon deprecation of `fetch_by_stable_id()` from the `MemberAdaptors`.

**Related JIRA tickets:**
- ENSCOMPARASW-5361

## Overview of changes
- Included GenomeDB_Adaptors and GenomeDB to replace `MemberAdaptor::fetch_by_stable_id($gene->stable_id)` with `MemberAdaptor::fetch_by_stable_id_GenomeDB($gene->stable_id, $genome)` in most/all scripts there abouts.
- Some small script repairs - which are easy to see.
- 
## Testing
These are example scripts - they do not require specific testing, however a local `prove -r modules/t` and Travis says nothing is broken.

## Notes
These scripts do not have to work, they are examples for API/method use.

### CODE REVIEW

File: scripts/examples/gene_gain_loss_tree_printAllNodes.pl [hunk 1]
```diff
@@ -33,16 +33,18 @@
...
-my $genome_db_adaptor   = $reg->get_adaptor ("Multi", "compara", "GenomeDB");
-
-my $member = $gene_member_adaptor->fetch_by_stable_id($gene_stable_id);
+my $genome_name = "homo_sapiens";
+
+my $gene_member_adaptor = $reg->get_adaptor("Multi", "compara", "GeneMember");
+my $gene_tree_adaptor   = $reg->get_adaptor("Multi", "compara", "GeneTree");
+my $cafe_tree_adaptor   = $reg->get_adaptor("Multi", "compara", "CAFEGeneFamily");
+my $genome_db_adaptor   = $reg->get_adaptor("Multi", "compara", "GenomeDB");
+print $cafe_tree_adaptor;
```
Comment (dthybert) [inline scripts/examples/gene_gain_loss_tree_printAllNodes.pl:43]: is this print necessary?

File: scripts/examples/gene_gain_loss_tree_printAllNodes.pl [hunk 2]
```diff
@@ -33,16 +33,18 @@
...
+print $genome_db_adaptor;
```
Comment (dthybert) [inline scripts/examples/gene_gain_loss_tree_printAllNodes.pl:44]: is this print necessary?

File: scripts/examples/gene_gain_loss_tree_printAllNodes.pl [hunk 3]
```diff
@@ -33,16 +33,18 @@
...
+my $genome = $genome_db_adaptor->fetch_by_name_assembly($genome_name);
+print $genome->genome_db_id, "\n";
```
Comment (dthybert) [inline scripts/examples/gene_gain_loss_tree_printAllNodes.pl:46]: is this print necessary?

Comment (CristiGuijarro) [inline scripts/examples/gene_gain_loss_tree_printAllNodes.pl:46]:
```suggestion
```

**Review by dthybert — APPROVED** [review]

**Pull request merged**